portfolio/src/main/webapp/script.js

```diff
@@ -114,19 +114,155 @@ function createCommentElement(server_comment) {
  * @param comment: a FormSubmissions instance stored in the server
  */
 function deleteComment(comment) {
-    const params = new URLSearchParams();
-    params.append('id', comment.id);
-    fetch('/delete-comment', {method: 'POST', body:params});
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-comment', {method: 'POST', body:params});
 }
 
 /** Tells the server to fetch and delete all comments. */
 function deleteAllComments() {
-    fetch('/data').then(response => response.text()).then((comments) => {
-      const commentsElement = document.getElementById('comments-container');
-      const commentsObj = JSON.parse(comments);
-      commentsObj.forEach((comment) => {
-        deleteComment(comment);
-      })
-      commentsElement.innerText = '';
-    });
-}
\ No newline at end of file
+  fetch('/data').then(response => response.text()).then((comments) => {
+    const commentsElement = document.getElementById('comments-container');
+    const commentsObj = JSON.parse(comments);
+    commentsObj.forEach((comment) => {
+      deleteComment(comment);
+    })
+    commentsElement.innerText = '';
+  });
+}
+
+/**
+ * Creates a map and adds it to the page.
+ */
+var purpleStyledMap;
+var map;
+var marker;
+function initMap() {
+  // Create a new StyledMapType object, passing it an array of desaturated purple
+  // styles, and the name to be displayed on the map type control. 
+  purpleStyledMap = new google.maps.StyledMapType(
```

Comment:
would be good to extract this out to a new method (ur method is going to be large)

Reply:
Right, split code for different APIs into different files.